### PR TITLE
Add support for `string | RegExp` to `ignoreValues` option of `value-no-vendor-prefix`

### PR DESCRIPTION
**Pull request by Mouvedia**
> Which issue, if any, is this issue related to?

see exceptions section of https://github.com/stylelint/stylelint/pull/7967#issue-2489168630

> Is there anything in the PR that needs further explanation?

```diff
-               'value-no-vendor-prefix': CoreRule<true, { ignoreValues: Array<StringOrRegex> }>;
+               'value-no-vendor-prefix': CoreRule<true, { ignoreValues: OneOrMany<StringOrRegex> }>;
```

### CODE REVIEW

File: lib/rules/value-no-vendor-prefix/index.mjs [hunk 1]
```diff
@@ -48,17 +45,21 @@ const rule = (primary, secondaryOptions) => {
...
 
 		/**
-		 * @type {{ raw: IgnoreValues, unprefixed: IgnoreValues }}
+		 * @typedef {string | RegExp} IgnoreValue
+		 * @type {{ raw: IgnoreValue[], unprefixed: IgnoreValue[] }}
 		 * @todo use Object.groupBy once Node v20 support is dropped
 		 */
 		const groups = { raw: [], unprefixed: [] };
-
-		secondaryOptions?.ignoreValues?.forEach((value) => {
```
Comment (ybiquitous) [inline lib/rules/value-no-vendor-prefix/index.mjs:56]: [suggest] The idiom using `flat()` seems suitable here:

```diff
-	secondaryOptions?.ignoreValues?.forEach((value) => {
+	[secondaryOptions?.ignoreValues ?? []].flat().forEach((value) => {
```

File: lib/rules/value-no-vendor-prefix/index.mjs [hunk 2]
```diff
@@ -48,17 +45,21 @@ const rule = (primary, secondaryOptions) => {
...
+		const ignoreValues = secondaryOptions?.ignoreValues;
+		const fillGroups = (/** @type {IgnoreValue} */ value) => {
 			const useRawValue = isRegExp(value) || (value.startsWith('/') && value.match(/\/i?$/));
```
Comment (ybiquitous) [inline lib/rules/value-no-vendor-prefix/index.mjs:55]: [note] Out-of-scope. We want to unify the similar logic with another opportunity.

https://github.com/stylelint/stylelint/blob/84f13fea65658a0ec6916567086ae73e8f77af00/lib/utils/matchesStringOrRegExp.mjs#L74-L78

**Review by ybiquitous — APPROVED** [review]
Thanks, overall LGTM. I've left one refactoring suggestion.

File: lib/rules/value-no-vendor-prefix/index.mjs (hunk 1 above)
Comment (Mouvedia) [reply lib/rules/value-no-vendor-prefix/index.mjs:56]: Just to be sure, you mean at line 62?

Comment (ybiquitous) [reply lib/rules/value-no-vendor-prefix/index.mjs:56]: No, I think the change above is only necessary and `if...else if` at line 61-62 is unnecessary.

Comment (ybiquitous) [reply lib/rules/value-no-vendor-prefix/index.mjs:56]: I think the following works:

```js
[secondaryOptions?.ignoreValues ?? []].flat().forEach((value) => {
	const useRawValue = isRegExp(value) || (value.startsWith('/') && value.match(/\/i?$/));
	const group = useRawValue ? 'raw' : 'unprefixed';

	groups[group].push(value);
});
```

Comment (Mouvedia) [reply lib/rules/value-no-vendor-prefix/index.mjs:56]: So no type check, I get it.

Comment (Mouvedia) [reply lib/rules/value-no-vendor-prefix/index.mjs:56]: It needs some tweaking, but I see what you mean.

**Pull request merged**